runtime/jcl/common/java_lang_Class.cpp

```diff
@@ -55,10 +55,12 @@ typedef enum {
 #define 	STACK_WALK_STATE_LIMITED_DOPRIVILEGED		(void *)2
 #define 	STACK_WALK_STATE_FULL_DOPRIVILEGED		(void *)3
 
+#if JAVA_SPEC_VERSION < 24
```

Comment:
`ObjsArraySizeNindex`, `STACK_WALK_STATE_LIMITED_DOPRIVILEGED` and `STACK_WALK_STATE_FULL_DOPRIVILEGED` can be removed for JDK24+.